File: api/start.py
import json
import os

def handler(request):
    try:
        body = request.get("body")
        if isinstance(body, str):
            body = json.loads(body)

        topic = body.get("topic")
        level = body.get("level")
        sub_category = body.get("sub_category")

        if topic in ["grammar", "vocab", "idioms"]:
            filename = f"{topic}{sub_category}_{level}.json"
        else:
            filename = f"{topic}_{level}.json"

        filepath = os.path.join("data", filename)

        if not os.path.exists(filepath):
            return {
                "statusCode": 404,
                "body": f"파일 {filename}이 존재하지 않습니다."
            }

        with open(filepath, "r", encoding="utf-8") as f:
            questions = json.load(f)

        return {
            "statusCode": 200,
            "headers": {"Content-Type": "application/json"},
            "body": json.dumps(questions)
        }

    except Exception as e:
        return {
            "statusCode": 500,
            "body": f"오류 발생: {str(e)}"
        }

File: api/start.mjs
import fs from "fs";
import path from "path";

export default function handler(req, res) {
  try {
    if (req.method !== "POST") {
      return res.status(405).json({ error: "Method not allowed" });
    }

    const body = req.body;
    const { topic, level, sub_category } = body;

    let filename;
    if (["grammar", "vocab", "idioms"].includes(topic)) {
      filename = `${topic}${sub_category}_${level}.json`;
    } else {
      filename = `${topic}_${level}.json`;
    }

    const filepath = path.join(process.cwd(), "data", filename);

    if (!fs.existsSync(filepath)) {
      return res.status(404).send(`파일 ${filename}이 존재하지 않습니다.`);
    }

    const questions = JSON.parse(fs.readFileSync(filepath, "utf-8"));
    res.status(200).json(questions);
  } catch (err) {
    res.status(500).send(`오류 발생: ${err.message}`);
  }
}
